### PR TITLE
Converted 'leave_channel' to typescript.

### PR DESCRIPTION
**Pull request by vraravam**
#### Summary
Converted `leave_channel` code to typescript

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/16140

### CODE REVIEW

File: components/channel_header_dropdown/menu_items/leave_channel/leave_channel.test.tsx [hunk 1]
```diff
@@ -47,10 +47,10 @@ describe('components/ChannelHeaderDropdown/MenuItem.LeaveChannel', () => {
         };
         const makeWrapper = () => shallow(<LeaveChannel {...props}/>);
 
-        props.channel.type = Constants.DM_CHANNEL;
```
Comment (devinbinnie) [inline components/channel_header_dropdown/menu_items/leave_channel/leave_channel.test.tsx:50]: Did these error out before you changed them? Might just need to clean up our constants a bit at some point.
Nothing to change now.

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @vraravam!

File: components/channel_header_dropdown/menu_items/leave_channel/leave_channel.test.tsx (hunk 1 above)
Comment (vraravam) [reply components/channel_header_dropdown/menu_items/leave_channel/leave_channel.test.tsx:50]: Yes - those errored out. And yes, once all code has been converted into ts, we will need to run a cleanup of things like these. Maybe a good idea to mark these with a `TechDebt` TODO?

**Review by nickmisasi — APPROVED** [review]
LGTM, thanks @vraravam

**Comment by jgilliam17** [comment]
Thank you @vraravam 
I am seeing some errors when leaving a private channel. Upon checking at least one is also found on master.
(New issue created for follow up - MM-30977)
I am also seeing an additional error here on the PR server that I am not able to reproduce on master - it shows up when a member is added to private channel and system message posts `you were added to channel...` 
Both appear to be related to Incident plugin. 
Apologize in advance, as there may be a delay in getting back to this PR due to holiday in USA. I will discus with the team on Tuesday and retest. Thank you.

Screen Shot 2020-11-25 at 5 03 13 PM

**Review by jgilliam17 — APPROVED** [review]
Thank you @vraravam 
Tested, looks good to merge.
- Verified leave channel works as expected.
Confirmed errors were related to incident plugin, filed additional issue for followup 
https://mattermost.atlassian.net/browse/MM-31007
@nickmisasi Can you please help merge? Thanks.

**Pull request merged**